search/models.py

```diff
@@ -0,0 +1,120 @@
+from django.db import models
+from django.db.backends.signals import connection_created
+from django.db.models.signals import class_prepared, post_save, pre_delete
+from django.dispatch import receiver
+from django.utils.module_loading import import_string
+
+from .utils import rank
+
+
+class Match(models.Lookup):
+    lookup_name = 'match'
+
+    def as_sql(self, qn, connection):
+        lhs, lhs_params = self.process_lhs(qn, connection)
+        rhs, rhs_params = self.process_rhs(qn, connection)
+        params = lhs_params + rhs_params
+        return '%s MATCH %s' % (lhs, rhs), params
+
+
+class SearchField(models.Field):
+    pass
+SearchField.register_lookup(Match)
+
+
+class SearchQuerySet(models.QuerySet):
+    def order_by_relevancy(self):
+        extra = {'relevancy': 'rank(matchinfo(idx))'}
+        return self.extra(select=extra).order_by('-relevancy')
+
+
+class Search(models.Model):
+    """Model that handle the search."""
+    rowid = models.IntegerField(primary_key=True)
+    model = models.CharField(max_length=64)
+    model_id = models.IntegerField()
+    public = models.BooleanField(default=True)
+    text = SearchField()
+
+    objects = SearchQuerySet.as_manager()
+
+    class Meta:
+        db_table = 'idx'
+        managed = False
+
+    @classmethod
+    def ids(cls, **kwargs):
+        qs = Search.objects.filter(**kwargs).order_by_relevancy()
+        return qs.values_list('model_id', flat=True)
+
+    @classmethod
+    def search(cls, **kwargs):
+        qs = Search.objects.filter(**kwargs).order_by_relevancy()
+        for row in qs:
+            yield _SEARCHABLE[row.model].objects.get(pk=row.model_id)
+
+
+class SearchMixin(models.Model):
+    """Inherit from this mixin to make your model searchable."""
+
+    class Meta:
+        abstract = True
+
+    @property
+    def index_strings(self):
+        return []
+
+    @property
+    def index_public(self):
+        return True
+
+    def is_indexable(self):
+        return True
```

Comment:
What is that for? Do you think you'll have models inheriting from `SearchMixin` but aren't indexable?